### DISCUSSION TITLE
RTL languages support

### DISCUSSION THREAD

**Discussion by orgolan** [strapi/strapi#6063] [Legacy Topics]
is there any support for hebrew or arabic?

**Comment by lauriejim** [comment]
Hello! Can you please precise your question, I'm not sure to understand what you are talking about.

**Reply by AminAddow** [reply]
@orgolan might mean right to left support in the strapi admin console. In the console, arabic or hebrew text added to the rich text format is not displayed right to left but left to right as in english. This makes writing text in the input field a bit more frustrating. 

![image](https://user-images.githubusercontent.com/32101833/90042659-4312d180-dccb-11ea-9be7-e7c30d616f46.png)

**Reply by derrickmehaffy** [reply]
@soupette didn't we have an option in the WYSIWYG for languages that require this like Japanese? I could of swore we had something configured per language in the code but can't find the reference.

**Reply by soupette** [reply]
@derrickmehaffy it's handled automatically only in the WYSIWYG since it's using [draftjs](https://draftjs.org/docs/advanced-topics-text-direction/).

**Reply by derrickmehaffy** [reply]
So it's something with DraftJS and not us?

**Reply by soupette** [reply]
The RTL support with the WYSIWYG seems to be working, I mean at least on my side.

**Reply by AminAddow** [reply]
@soupette Can you attach a screenshot of how this looks in the admin page?

**Comment by desyatnik** [comment]
Well, for me it doesn't work for sure.
![image](https://user-images.githubusercontent.com/34752947/93201826-61e90580-f75a-11ea-996d-3939130ca150.png)

Is there any setting I could apply to align it to the right?

**Reply by desyatnik** [reply]
As a matter of update, I was able to resolve the issue by importing the original CSS from Draft.js following this suggestion: https://github.com/strapi/strapi/issues/3114#issuecomment-629813215

**Reply by desyatnik** [reply]
Oh well, it only works in markdown mode, but switches back in preview one. 
Could you please point me into the correct direction what component should I overwrite to tackle the preview mode? 
(adjusting PreviewWysiwyg didn't do much)

**Comment by jalal246** [comment]
In `extensions\content-manager\admin\src\components\WysiwygEditor\index.js`

I added 

```js
import React from "react";
import { Editor } from "draft-js";
import "draft-js/dist/Draft.css"; // only add this line
import PropTypes from "prop-types";

class WysiwygEditor extends React.Component {
  render() {
    return (
      <Editor textAlignment="right" {...this.props} ref={this.props.setRef} />
    );
  }
}

WysiwygEditor.defaultProps = {
  setRef: () => {},
};

WysiwygEditor.propTypes = {
  setRef: PropTypes.func,
};

export default WysiwygEditor;

```

It works, but as @desyatnik pointed out. it doesn't work for preview mode.
